### ISSUE TITLE
Custom Units Checker qualifiers need to be supertypes of @UnitsBottom

### ISSUE THREAD

**Issue by wmdietl** [typetools/checker-framework#2087]
Problem report: https://groups.google.com/d/msg/checker-framework-discuss/M0tPOCT9RGA/aRETL3X3CQAJ

`@UnitsBottom` is the implicit qualifier for null: `@ImplicitFor(typeNames = Void.class, literals = LiteralKind.NULL)`.

For the built-in qualifiers, assigning `null` works, e.g. `@m Float x = null;`

However, for a custom units annotation, the same is not true and an error is raised:

```
error: [return.type.incompatible] incompatible types in return.
            return null;
                   ^
 found   : @UnknownUnits  null
 required: @frac(Prefix.deci) Float
```

Note that `@UnitsBottom` has the following meta-annotation: `@SubtypeOf({}) // needs to be done programmatically`
So most likely the error is that setting the correct supertypes only considered the built-in units and incorrectly ignores costum units annotations.
